node/src/test_utils/persistent_configuration_mock.rs

```diff
@@ -678,6 +678,8 @@ impl PersistentConfigurationMock {
 
     set_arbitrary_id_stamp_in_mock_impl!();
 
+    // TODO: Review this, maybe we should return an error instead of panic?
+    // Also unsure why we have the else if clause.
```

Comment:
To be honest, my opinion is that this utility could go fully away.
It causes more making assumptions than any real good. 
I like simple and predictable mocks personally. 
This bypass the normally expected simplicity.

Should be replaced by simple
```  ...
     self.certain_method_results.remove(0)
```

The problem is if you remove this junk, you'll inevitably get a whole bunch of tests failing. My prediction would be that in such tests most likely a single result would be provided for a particular method, but it used to be replicated. After the code removal, it will panic for lacking enough elements in the Vec of results. 

**Maybe better approach**

The original idea of this utility probably was not to have to load the mock with a couple of same values for methods which, in their nature, should return an identical value at every call. Such as the wallet getters (it'd be weird to think of them as a value that changes during the life of the Node). Of course, you can argue, that there is a short moment when it truly returns None or so, when they haven't been populated in the DB yet. 

It seems like that could be the reason for the last clause in the if statement. The `else` allows removing a multiple different values in a row as it happens in each normal mock. Therefore values like:

Call 1 --> None,
Call 2 --> None,
Call 3 --> Some(<YOUR-WALLET>)

So, the best way to address this would be to preserve this method, but hopefully with an altered name representing the purpose, or the same achieved through a comment. Then go visit all the places where it is used and answer for yourself if it is a parameter which is supposed to return the same value in like 95% of cases or more. If yes, leave it in place. If not, replace it by the code I sketched up up here before.